### PR TITLE
Add wrapping logic to batch edit notes.

### PR DESCRIPTION
**Pull request by mathewjordan**
# Summary 
This adds wrapping UI logic and processing to Notes and Related URL fields in the batch edit workflow. As `replace` was already working for these fields, I don't believe an back end work is required here to make this work as expected.

# Specific Changes in this PR
- Adds Replace, Append, and Delete options to Notes and Related URL fields
- Adds functionality to process editType options for Notes and Related URL fields

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.

- [x] Patch
- [ ] Minor
- [ ] Major

# Steps to Test

Test batch edit process for:

### Notes
- [x] Append
- [x] Delete
- [x] Replace

### Related URL
- [x] Append
- [x] Delete
- [x] Replace

Also please let developers know if there are any special instructions to test this in the development environment. 

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
  - [ ] Database Schema changes
  - [ ] GraphQL API
  - [ ] Elasticsearch API
  - [ ] Ingest Sheet
  - [ ] CSV metadata export/update API
  - [ ] Shared Links export API
- Backwards-incompatible API changes
  - [ ] Database Schema changes
  - [ ] GraphQL API
  - [ ] Elasticsearch API
  - [ ] Ingest Sheet
  - [ ] CSV metadata export/update API
  - [ ] Shared Links export API
- [ ] Requires data migration
- [ ] Requires database triggers disabled during deployment/migration
- [ ] Requires reindex
- [ ] Terraform changes
  - [ ] Adds/requires new or changed Terraform variables
- [ ] Pipeline configuration changes (requires `mix meadow.pipeline.setup` run)
- [ ] Requires new variable added to `miscellany`
- [ ] Specific deployment synchronization instructions with other apps/API's
- [ ] Other specific instructions/tasks

# Tested/Verified
- [ ] End users/stakeholders

### CODE REVIEW

**Comment by mbklein** [comment]
@mathewjordan Try rebasing on `deploy/staging` – there’s a cache key in the build spec that’s pointing to a bad cache.

**Review by adamjarling — CHANGES_REQUESTED** [review]
See screenshots on some stuff I'm seeing.  Note the "Reading Room" value showing up in the Batch Confirmation window happens on `deploy/staging` branch for me too.

But the other issue happens for me when:
1. I add either a Notes or Related URL value(2), and click to go to Confirmation modal.
2. In Confirmation modal, click "Cancel"
3. Remove the added items for Notes and/or Related URL.  Proceed to Confirmation modal again.
4. Notice if you chose to "Replace" these items, a record of them still exists in the Confirmation modal, even though they were deleted.

image

**Comment by mathewjordan** [comment]
> See screenshots on some stuff I'm seeing. Note the "Reading Room" value showing up in the Batch Confirmation window happens on `deploy/staging` branch for me too.
> 
> But the other issue happens for me when:
> 
>     1. I add either a Notes or Related URL value(2), and click to go to Confirmation modal.
> 
>     2. In Confirmation modal, click "Cancel"
> 
>     3. Remove the added items for Notes and/or Related URL.  Proceed to Confirmation modal again.
> 
>     4. Notice if you chose to "Replace" these items, a record of them still exists in the Confirmation modal, even though they were deleted.
> 
> 
> image

Thanks. Taking a look.

File: assets/js/components/BatchEdit/Tabs.jsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ import UITabsStickyHeader from "@js/components/UI/Tabs/StickyHeader";
 import { Button, Notification } from "@nulib/design-system";
 import UIIconText from "../UI/IconText";
 import { IconAlert } from "@js/components/Icon";
+import { forEach } from "lodash";
```
Comment (adamjarling) [inline assets/js/components/BatchEdit/Tabs.jsx:26]: Do we need to use `lodash`s version of `forEach` here?

**Review by adamjarling — COMMENTED** [review]
@mathewjordan Code looks solid and minimal.  Nice work.

**Comment by mathewjordan** [comment]
The Reading Room thing looks unrelated and introduced elsewhere. I'll write up an issue for this.

image

**Review by adamjarling — APPROVED** [review]
Looking good!

**Pull request merged**